### PR TITLE
[JENKINS-60449] Update plugins to illustrate a new case

### PR DESCRIPTION
**Pull request by Vlatombe**
https://issues.jenkins-ci.org/browse/JENKINS-60449

Used by https://github.com/jenkinsci/jenkins/pull/4393

### CODE REVIEW

File: depender/src/main/java/org/jenkinsci/plugins/dependencytest/depender/DependerExtension2.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Extension to test optional dependencies.
+ */
+@OptionalExtension(requirePlugins = {"dependee"})
+public class DependerExtension2 {
+    private static void foo(Dependee d) {
+
+    }
+}
```
Comment (Vlatombe) [inline depender/src/main/java/org/jenkinsci/plugins/dependencytest/depender/DependerExtension2.java]: This is the key change

File: depender/src/main/java/org/jenkinsci/plugins/dependencytest/depender/DependerExtension2.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License
+ * 
+ * Copyright (c) 2013 IKEDA Yasuyuki
```
Comment (ikedam) [inline depender/src/main/java/org/jenkinsci/plugins/dependencytest/depender/DependerExtension2.java]: You can write your name here.

**Comment by ikedam** [comment]
Relates:
* jenkinsci/jenkins#4393
* https://plugins.jenkins.io/variant
* https://github.com/Vlatombe/jenkins-50336
* jenkinsci/jenkins#3370

**Review by ikedam — COMMENTED** [review]
I don’t think it’s a good idea to have `depender` depends on a specific plugin as it illustrates a minimum dependency relationships to test.

And it looks strange to me that `OptionalExtension` isn’t provided as a part of Jenkins core though it controls plugin loadings and that the test code exists in Jenkins core.

How about adding a new module `optionaldepender`?

**Comment by Vlatombe** [comment]
I can move the new stuff to an optionaldepender module, sounds fair.

**Review by ikedam — APPROVED** [review]
Thanks for update.
Looks good to me.

**Pull request merged**